worlds/tunic/er_rules.py

```diff
@@ -1537,3 +1647,129 @@ def set_er_location_rules(world: "TunicWorld") -> None:
              lambda state: has_sword(state, player))
     set_rule(multiworld.get_location("Shop - Coin 2", player),
              lambda state: has_sword(state, player))
+
+    def combat_logic_to_loc(loc_name: str, combat_req_area: str, set_instead: bool = False,
+                            dagger: bool = False, laurel: bool = False) -> None:
+        # dagger means you can use magic dagger instead of combat for that check
+        # laurel means you can dodge the enemies freely with the laurels
+        if set_instead:
+            set_rule(multiworld.get_location(loc_name, player),
+                     # someome tell me if you actually need to do the p=player and c=combat_req_area, lambdas scary
+                     lambda state, p=player, c=combat_req_area, d=dagger, la=laurel:
+                     has_combat_reqs(c, state, p)
+                     or (state.has(ice_dagger, player) if d else False)
+                     or (state.has(laurels, player) if la else False))
+        else:
+            add_rule(multiworld.get_location(loc_name, player),
+                     lambda state, p=player, c=combat_req_area, d=dagger, la=laurel:
+                     has_combat_reqs(c, state, p)
+                     or (state.has(ice_dagger, player) if d else True)
+                     or (state.has(laurels, player) if la else False))
```

Comment:
There's no loops here, so this is fine (same for the next one). You could also simplify some of these:
```suggestion
                     # someome tell me if you actually need to do the p=player and c=combat_req_area, lambdas scary
                     lambda state:
                     has_combat_reqs(c, state, p)
                     or (state.has(ice_dagger, player) and d)
                     or (state.has(laurels, player) and la))
        else:
            add_rule(multiworld.get_location(loc_name, player),
                     lambda state:
                     has_combat_reqs(c, state, p)
                     or (state.has(ice_dagger, player) or d)
                     or (state.has(laurels, player) and la))
```